.github/workflows/build.yml

```diff
@@ -118,4 +118,23 @@ jobs:
           command: composer require "laravel/framework:${{ matrix.laravel-version }}.*" --no-update
 
       - name: Run tests
-        run: vendor/bin/phpunit --verbose
+        run: vendor/bin/phpunit --verbose --coverage-clover clover.xml
+
+      - name: Upload coverage to Coveralls
+        env:
+          COVERALLS_PARALLEL: true
+          COVERALLS_REPO_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
What is this token?

Reply:
Github's actions provide has an environment variable `GITHUB_TOKEN ` that generates an "auth-token" that lets `coveralls.io` use to "read" this public repo via Github API.  

https://docs.github.com/en/free-pro-team@latest/actions/reference/authentication-in-a-workflow#permissions-for-the-github_token

https://docs.github.com/en/free-pro-team@latest/rest/reference/permissions-required-for-github-apps#permission-on-actions

you can also use it in actions to interact with Github's API.
```
name: Create issue on commit
on:
- push
jobs:
  create_commit:
    runs-on: ubuntu-latest
    steps:
    - name: Create issue using REST API
      run: |
        curl --request POST \
        --url https://api.github.com/repos/${{ github.repository }}/issues \
        --header 'authorization: Bearer ${{ secrets.GITHUB_TOKEN }}' \
        --header 'content-type: application/json' \
        --data '{
          "title": "Automated issue for commit: ${{ github.sha }}",
          "body": "This issue was automatically created by the GitHub Action workflow **${{ github.workflow }}**. \n\n The commit hash was: _${{ github.sha }}_."
          }'
```